src/app/games/dvonn/DvonnPieceStack.ts

```diff
@@ -3,26 +3,12 @@ import { NumberEncoder } from 'src/app/jscaip/Encoder';
 
 export class DvonnPieceStack {
     public static sizeEncoder: NumberEncoder<number> = NumberEncoder.numberEncoder(49);
-    public static encoder: NumberEncoder<DvonnPieceStack> = new class extends NumberEncoder<DvonnPieceStack> {
-        public maxValue(): number {
-            return (DvonnPieceStack.sizeEncoder.maxValue() *
-                Player.numberEncoder.shift() * Player.numberEncoder.maxValue() *
-                NumberEncoder.booleanEncoder.shift() * NumberEncoder.booleanEncoder.maxValue());
-        }
-        public encodeNumber(stack: DvonnPieceStack): number {
-            return ((DvonnPieceStack.sizeEncoder.encodeNumber(stack.size) *
-                Player.numberEncoder.shift() + Player.numberEncoder.encodeNumber(stack.owner)) *
-                NumberEncoder.booleanEncoder.shift() + NumberEncoder.booleanEncoder.encodeNumber(stack.source));
-        }
-        public decodeNumber(encoded: number): DvonnPieceStack {
-            const sourceN: number = encoded % NumberEncoder.booleanEncoder.shift();
-            encoded = (encoded - sourceN) / NumberEncoder.booleanEncoder.shift();
-            const playerN: number = encoded % Player.numberEncoder.shift();
-            encoded = (encoded - playerN) / Player.numberEncoder.shift();
-            const size: number = encoded;
-            return new DvonnPieceStack(Player.of(playerN), size, NumberEncoder.booleanEncoder.decode(sourceN));
-        }
-    }
+    public static encoder: NumberEncoder<DvonnPieceStack> = NumberEncoder.ofCombination(
+        [NumberEncoder.booleanEncoder, Player.numberEncoder, DvonnPieceStack.sizeEncoder],
```

Comment:
NumberEncoder.booleanEncoder ça sonne spécial haha :D
Ce serait plus clair avec Hasher pour any -> number and Jsonifier pour any -> jsonAny ?

Reply:
Ça sonne spécial effectivement, me suis dit la même chose :d Mais je suis pas pas pour "hasher" comme nom, pour moi hash c'est one way, ici c'est bijectif.